include/internal/e_os.h

```diff
@@ -96,7 +96,7 @@
         * might be possible to achieve the goal by /DELAYLOAD-ing .DLLs
         * and check for current OS version instead.
         */
-#    define _WIN32_WINNT 0x0501
+#    define _WIN32_WINNT 0x0600
```

Comment:
My understanding is that  increasing `_WIN32_WINNT` allows us to use newer features, but it does not automatically break compatibility. The availability of WSARecvMsg is probed at runtime and there is a fallback if it is not available.

By bumping this there is a risk we accidentially start depending on APIs not supported on XP. I guess maybe we should support a build with version 0x0501, maybe in CI.

Reply:
My code as stands detects availability of WSARecvMsg-related definitions by whether WSAID_WSARECVMSG is defined. So we could make _WIN32_WINNT configurable and the code would automatically use the best function available, whether it's set to 0x0501 or 0x0600. We could discuss what we want to do about evolving _WIN32_WINNT, though it feels like a separate discussion.

Reply:
> we could make _WIN32_WINNT configurable

I think it should already be configurable, i.e. you can pass `-D_WIN32_WINNT=...` at configure time. Does your code require _WIN32_WINNT to be 0x0600 before it will work?

The change here changes the default value for `_WIN32_WINNT` which I don't think we can do without some agreement that it is ok to do so.

Reply:
I suppose we should document it in the NOTES.windows then? And possibly also change the default to 0x0600 if OMC agrees so (supposedly the override with 0x0501 should be kept working).

Reply:
I raised this at the OMC meeting today, and it led to this proposal:

https://github.com/openssl/general-policies/issues/22